spinalcordtoolbox/scripts/sct_run_batch.py

```diff
@@ -202,10 +222,29 @@ def run_single(subj_dir, script, script_args, path_segmanual, path_data, path_da
         'ITK_GLOBAL_DEFAULT_NUMBER_OF_THREADS': str(itk_threads),
         'SCT_PROGRESS_BAR': 'off'
     })
+    if 'SCT_DIR' not in envir:  # For native Windows installations, the install script won't add SCT_DIR
+        envir['SCT_DIR'] = __sct_dir__
+
+    cmd = [script_full, subj_dir] + script_args.split(' ')
+
+    # Make sure that a Windows-compatible bash port is used to run shell scripts
+    if sys.platform == "win32":
+        with open(script_full) as f:
+            shebang = f.readline()
+        if script_full.endswith('.sh') or "sh" in shebang:
```

Comment:
Thanks for pointing this out!! How about the `^#!.*\/.*sh\b.*` regex?

![image](https://user-images.githubusercontent.com/16181459/160473059-ef350ac1-be2f-464a-ba7b-4ea610b2d3df.png)

Reply:
It's very much a matter of personal taste, but since you asked: I'm partial to `re.match(r'#!\S*sh\b', first_line, re.ASCII)`, personally, because
* `re.match()` looks at the start of the string, hence no initial `^`.
* `/` isn't technically required?
* `.*` matches whitespace, but I think we want the first word to end in `sh`, so `\S*` is more accurate.
* I just checked in my terminal, and non-ascii whitespace seems to be non-word-splitting, hence the `re.ASCII` flag.